actors/builtin/miner/miner_actor.go

```diff
@@ -1572,6 +1573,26 @@ func (a Actor) WithdrawBalance(rt Runtime, params *WithdrawBalanceParams) *adt.E
 	return nil
 }
 
+func (a Actor) RepayDebt(rt Runtime, params *adt.EmptyValue) *adt.EmptyValue {
+	var st State
+	var fromVesting, fromBalance abi.TokenAmount
+	rt.State().Transaction(&st, func() {
+		var err error
+		info := getMinerInfo(rt, &st)
+		rt.ValidateImmediateCallerIs(append(info.ControlAddresses, info.Owner, info.Worker)...)
+
+		// Verify unlocked funds cover both InitialPledgeRequirement and FeeDebt
```

Comment:
This first part of the comment doesn't make sense after @ZenGround0 's recent changes.  I don't think this will or should fail of only part of the debt can be paid off.